### ISSUE TITLE
Unify arguments of `__new__` and `__init__`

### ISSUE THREAD

**Issue by smurfix** [PyCQA/pylint#3718]
Consider this code:
```
class Foo:
    def __new__(cls, a,b,c):
        obj = object.__new__(cls)
        do_something_with(obj,a,b)
        obj._x = 123
    def __init__(self, a,b):
        do_something_else_with(b)
```
There are four somewhat-related problems here:

* `__new__` doesn't return anything, but pylint doesn't complain.

* the arguments of `__new__` and `__init__` are clearly incompatible, but pylint doesn't notice.

* pylint complains about the unused argument `a` in `__init__`. It shouldn't, `a` is used in `__new__`. The fact that `__init__` does not also use it is immaterial and I don't want to clutter my code with a pylint-disable comment for it.

* Also in that kind of code it's common to initialize attributes inside `__new__`. pylint should record them and not complain with `attribute-defined-outside-init`. The object for which to record attributes is whatever `__new__` returns, which is another reason to check that it in fact returns *something*. ;-)

**Comment by AWhetter** [comment]
Because this is reporting several issues, please could you post an issue for each individual problem you are having? It's difficult to track what is completed and when isn't when an issue reports multiple problems.
